mocks/DDrppi_mocks/countpairs_rp_pi_mocks_impl.c.src

```diff
@@ -114,63 +114,66 @@ countpairs_mocks_func_ptr_DOUBLE countpairs_rp_pi_mocks_driver_DOUBLE(const stru
 
     /* Array of function pointers */
     countpairs_mocks_func_ptr_DOUBLE allfunctions[] = {
-#ifdef __AVX512F__
-      countpairs_rp_pi_mocks_avx512_intrinsics_DOUBLE,
+        countpairs_rp_pi_mocks_fallback_DOUBLE,
+#ifdef __SSE4_2__
+        countpairs_rp_pi_mocks_sse_intrinsics_DOUBLE,
 #endif
 #ifdef __AVX__
-      countpairs_rp_pi_mocks_avx_intrinsics_DOUBLE,
+        countpairs_rp_pi_mocks_avx_intrinsics_DOUBLE,
 #endif
-#ifdef __SSE4_2__
-      countpairs_rp_pi_mocks_sse_intrinsics_DOUBLE,
+#ifdef __AVX512F__
+        countpairs_rp_pi_mocks_avx512_intrinsics_DOUBLE,
 #endif
-      countpairs_rp_pi_mocks_fallback_DOUBLE
     };
 
     const int num_functions = sizeof(allfunctions)/sizeof(void *);
-    const int fallback_offset = num_functions - 1;
+    const int fallback_index = 0;
 #if defined(__AVX512F__) || defined(__AVX__) || defined(__SSE4_2__)
-    const int highest_isa = instrset_detect();
+    const int highest_isa = get_max_usable_isa();
 #endif
-    int curr_offset = 0;
+    int curr_index = 0;
 
-    /* Check for AVX512F support */
-    int avx512_offset = fallback_offset;
-#ifdef __AVX512F__
-    avx512_offset = highest_isa >= 9 ? curr_offset:fallback_offset;
-    curr_offset++;
+    /* Is the SSE function supported at runtime and enabled at compile-time?*/
+    int sse_index = curr_index;
+#ifdef __SSE4_2__
+    curr_index++;
+    if(highest_isa >= SSE42)
+        sse_index = curr_index;
 #endif
 
     /* Now check if AVX is supported by the CPU */
-    int avx_offset = fallback_offset;
+    int avx_index = curr_index;
 #ifdef __AVX__
-    avx_offset = highest_isa >= 7 ? curr_offset:fallback_offset;
-    curr_offset++;
+    curr_index++;
+    if(highest_isa >= AVX)
```

Comment:
If you are going to avoid the curly braces, then I would recommend putting the entire `if` in a single line. Otherwise, use the opening and closing braces